service/dap/server.go

```diff
@@ -378,6 +385,8 @@ func (s *Server) recoverPanic(request dap.Message) {
 
 func (s *Server) handleRequest(request dap.Message) {
 	defer s.recoverPanic(request)
+	s.resumeMu.Lock()
```

Comment:
The PR comment seems to indicate that you don't want to interrupt finishing next with a halt. But this would block all other requests as well, including disconnect.

Reply:
I understand now that you will only block while auto-continue is starting up, akin to blocking with resumeRequestLoop when starting up async requests.